bluesky/tests/test_external_assets_and_paging.py

```diff
@@ -0,0 +1,433 @@
+from typing import Dict, Iterator
+from bluesky import Msg
+from event_model.documents.event_descriptor import DataKey
+from event_model.documents.event_page import PartialEventPage
+from event_model.documents.event import PartialEvent
+from event_model.documents.resource import PartialResource
+from event_model.documents.datum import Datum
+from event_model.documents.stream_datum import StreamDatum
+from event_model.documents.stream_resource import StreamResource
+from bluesky.utils import new_uid, IllegalMessageSequence
+from itertools import chain
+import pytest
+from bluesky.protocols import (
+    Asset, Readable,
+    Reading, SyncOrAsync, WritesExternalAssets, Flyable, EventCollectable, EventPageCollectable, Collectable
+)
+
+
+def read_Readable(self) -> Dict[str, Reading]:
+    return dict(det2=dict(value=1.2, timestamp=0.0))
+
+
+def describe_Readable(self) -> Dict[str, DataKey]:
+    return dict(
+        det1=dict(source="hw1", dtype="number", shape=[], external="STREAM:"),
+        det2=dict(source="hw2", dtype="number", shape=[])
+    )
+
+
+def collect_asset_docs_Resource(self) -> Iterator[Asset]:
+    resource = PartialResource(
+        resource_kwargs={"some_argument": 1337},
+        root="awesome",
+        spec=".dat",
+        resource_path="no/really,/awesome",
+        uid=new_uid(),
+    )
+
+    yield "resource", resource
+
+
+def collect_asset_docs_Datum(self) -> Iterator[Asset]:
+    datum = Datum(
+        datum_id="some_resource_name/123123",
+        resource="uid_of_that_resource",
+        datum_kwargs={"some_argument": 1234},
+    )
+    yield "datum", datum
+
+
+def collect_asset_docs_StreamDatum(self) -> Iterator[Asset]:
```

Comment:
Please can we also add a function that gives 2 StreamResources and StreamDatums. This is what a flyer will do, proxying Stream documents from 2 detectors. Or even a single detector making 2 datasets